document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/vistors/PostgresFilterTypeExpressionVisitor.java

```diff
@@ -81,16 +96,108 @@ public String visit(final KeyExpression expression) {
+        final String parsedLhs = expression.getArraySource().accept(arrayPathVisitor);
+
+        // Extract the field name
+        final String identifierName =
+            expression
+                .getArraySource()
+                .accept(new PostgresIdentifierExpressionVisitor(postgresQueryParser));
+
+        // If the field name is 'elements.inner', just pick the last part as the alias ('inner')
+        final String alias = getLastPath(identifierName);
+
+        // Any LHS field name (elements) is to be prefixed with current alias (inner)
+        final PostgresWrappingFilterVisitorProvider visitorProvider =
+            new PostgresArrayRelationalWrappingFilterVisitorProvider(
+                postgresQueryParser, identifierName, alias);
+        final String parsedFilter =
+            expression
+                .getFilter()
+                .accept(
+                    new PostgresFilterTypeExpressionVisitor(postgresQueryParser, visitorProvider));
+
+        return String.format(
+            "EXISTS (SELECT 1 FROM jsonb_array_elements(COALESCE(%s, '[]'::jsonb)) AS \"%s\" WHERE %s)",
+            parsedLhs, alias, parsedFilter);
+
+      default:
+        throw new UnsupportedOperationException(
+            "Unsupported array operator: " + expression.getOperator());
+    }
   }
 
-  @SuppressWarnings("unchecked")
+  @SuppressWarnings({"unchecked", "SwitchStatementWithTooFewBranches"})
   @Override
   public String visit(final DocumentArrayFilterExpression expression) {
-    throw new UnsupportedOperationException();
+    /*
+    EXISTS
+    (SELECT 1
+     FROM  jsonb_array_elements(COALESCE(document->'planets', '[]'::jsonb)) AS planets
+     WHERE <parsed_containing_filter_with_aliased_field_names>
+     )
+     */
+    switch (expression.getOperator()) {
+      case ANY:
+        // Convert 'elements' to planets->'elements' where planets could be an alias for an upper
+        // level array filter
+        // Also, for the first time (if this was not under any nesting), use the field identifier
+        // visitor to make it document->'elements'
+        final PostgresIdentifierExpressionVisitor identifierVisitor =
+            new PostgresIdentifierExpressionVisitor(postgresQueryParser);
+        final PostgresSelectTypeExpressionVisitor arrayPathVisitor =
+            wrappingVisitorProvider == null
+                ? new PostgresFieldIdentifierExpressionVisitor(identifierVisitor)
+                : wrappingVisitorProvider.getForNonRelational(identifierVisitor);
+        final String parsedLhs = expression.getArraySource().accept(arrayPathVisitor);
+
+        // Extract the field name
+        final String identifierName =
+            expression
+                .getArraySource()
+                .accept(new PostgresIdentifierExpressionVisitor(postgresQueryParser));
+
+        // If the field name is 'elements.inner', just pick the last part as the alias ('inner')
```

Comment:
Could this be an issue if there are two array expressions, both having the same last part, such as `inner` in `planets.inner` and `elements.inner`?

If so, creating an alias with the full path, like `elements_inner`, might be a helpful solution.

Reply:
Actually, the alias is used within a smaller scope, so it shouldn't be an issue. But, yeah, it still makes sense to update it for clarity. Will do.